src/jit/codegenxarch.cpp

```diff
@@ -5406,7 +5406,12 @@ void CodeGen::genJmpMethod(GenTreePtr jmp)
         // assert should hold.
         assert(varDsc->lvRegNum != REG_STK);
 
-        var_types loadType = varDsc->lvaArgType();
+        if (varDsc->lvIsStructField)
```

Comment:
Nit: you could write this as `assert(!varDsc->lvIsStructField || (compiler->lvaTable[varDsc->lvParentLcl].lvFieldCnt == 1))`